Automattic/wp-calypso
# Extend "Whats new" modal for global nav

This PR is currently to demonstrate how the what's new modal can be extended to show the new content tour as described in: pfsHM7-cD-p2

This solves the technical challenge of loading the what's new modal without displaying the help center. Previously, it could only be shown if the help center slide out menu is also visible.

It also restyles the popup to be aligned to the bottom right and not interupt using calypso as much.

<img width="1728" alt="Screenshot 2024-02-26 at 5 04 28 pm" src="https://github.com/Automattic/wp-calypso/assets/22446385/30eca058-4822-46c3-8b01-e3c483b8d22f">

I'm adding logic that will keep track of which announcements the user has seen at a new endpoint `/whats-new/seen-announcement-ids` in D139800-code.

### Testing instructions
Apply D139800-code to your sandbox, and this PR.
Filter the network tab to `whats-new`
You should be able to see the modal initially, close it, and reopen it from "What's New" in the help menu

<img width="416" alt="Screenshot 2024-02-26 at 5 07 22 pm" src="https://github.com/Automattic/wp-calypso/assets/22446385/32750506-29d9-44d4-ad89-9472d8dfd982">

The whats new tour is also available within the editor via the ETK, this code had to be modified as well.

### To do in a follow up PR.
* Probably clean out old code relating to `HasSeenWhatsNewModal` which isn't useful. I see there is code for an indicator dot, I should be able to get this working as it was originally intended using the new `latestSeenWhatsNewModalItem` counter.
* Potentially change from using the older redux store pattern to a mutation?
* Confirm that the ETK whats-new javascript package is unused.
